pkg/operator/ceph/cluster/mgr/spec.go

```diff
@@ -90,7 +100,18 @@ func (c *Cluster) makeDeployment(mgrConfig *mgrConfig) *apps.Deployment {
 
 	replicas := int32(1)
 
-	d := &apps.Deployment{
+	d.Spec = apps.DeploymentSpec{
+		Selector: &metav1.LabelSelector{
+			MatchLabels: c.getPodLabels(mgrConfig.DaemonID),
+		},
+		Template: podSpec,
+		Replicas: &replicas,
+		Strategy: apps.DeploymentStrategy{
+			Type: apps.RecreateDeploymentStrategyType,
+		},
+	}
+
+	/*d := &apps.Deployment{
```

Comment:
We can restore this now if we just return nil above

Reply:
Done.